joschahenningsen/TUM-Live
# Protect vod access with signed jwt

### Motivation and Context



Currently any url to a vod can be 
- guessed
- shared

We want TUM-Live to determine whether a user is allowed to watch a video and only then generate a URL for them to watch it.

### Description


This PR adds 
- A way for the edge server to acquire the public jwt signing key of TUM-Live via the path `/jwtPubKey`
- A way for TUM-Live to issue a jwt for a user with these claims:
  - `userID`: If a user shared a URL we know who it was
  - `exp`: a playlist URL is valid for 7 hours after issuing
  - `playlist_URL`: The path of the video the user is allowed to watch, e.g. `https://edge.tum.live/abc.mp4/playlist.m3u8` 
  - Not yet implemented but worth a look: `userIP`: The IP of the viewer who is allowed to watch the video, if the URL was shared with someone on another network it becomes useless.  We could also use the ISP instead.
- A way for the edge server to reject and approve requests based on the validity of the jwt and its claims.

### Steps for Testing

Prerequisites:
- TUM-Live running
- vod-service running
- 1 worker running
- 1 edge server running

1. upload a vod
2. watch the vod and try playing with the url, the video should not play if the url was modified

### Screenshots
